manifests/configfile.pp

```diff
@@ -42,6 +42,7 @@
 # @author https://github.com/elastic/puppet-logstash/graphs/contributors
 #
 define logstash::configfile (
+  $ensure = $::logstash::ensure,
```

Comment:
If you think this parameter is required, please add some documentation above:

```
# @param ensure
#   ...
```

But AFAIAC, i think we do not need this extra parameter and we can directly use `logstash::ensure`. In this case, this line can be removed, and the two bellow changes are required:

Reply:
i think that more granularity (with proper default values) is always better
in this case, we had a configfile that was placed by this module, then later i wanted to remove it so i made these changes and specifically called `absent`

could i have accomplished the same with a `file` resource? yes ofc! but we place the files in `/etc/logstash/conf.d` with `logstash::configfile` so we should be able to remove them the same way

alternatively (or probably?) we could/should manage the entire `/etc/logstash/conf.d` directory (with `purge => true` so that no files are leftover